lib/core/src/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobin.hs

```diff
@@ -1375,6 +1495,196 @@ makeChangeForCoin
+-- the last. This property is used by "changeMapOutputCoinPairs"
+-- so it's important it's maintained.
+--
+-- The following two functions work by modifying the change bundles for
+-- non-user-specified assets by adding minted tokens to the largest change
+-- bundle:
+--
+-- [ [          ("B", 7)  ]
+-- [ [("A", 1), ("B", 8)  ]
+-- [ [("A", 2), ("B", 9)  ]
+-- [ [("A", 3), ("B", 9)  ]
+-- [ [("A", 4), ("B", 12) ] <-- add mint tokens here
+--
+-- And to remove burnt tokens from the smallest change bundles
+-- until all required burnt tokens are removed:
+--
+-- [ [          ("B", 7)  ] <-- start removing burnt tokens here
+-- [ [("A", 1), ("B", 8)  ]  <-- if need to burn more, remove here
+-- [ [("A", 2), ("B", 9)  ]    <-- and so on...
+-- [ [("A", 3), ("B", 9)  ]
+-- [ [("A", 4), ("B", 12) ]
+--
+-- The solution for minting maintains the properties we desire, namely:
+--   1. The number of change bundles matches the number of
+--      "outputs to cover" (we are not changing the number of bundles).
+--   2. The change bundles are partially ordered in ascending order (by
+--      adding to the largest bundle we trivially maintain ordering).
+--   3. The change bundles are split in such a way to maximize the
+--      number of large change bundles. By adding to the largest, we are
+--      only increasing the "usefulness" of that bundle.
+--
+-- The solution for burning maintains the same properties:
+--   1. The number of change bundles is not changed, in the case we burn a
+--      change bundle completely, we just leave it as an empty entry
+--      (effectively "pad with zeros").
+--   2. By removing from the smallest bundle, we maintain the ascending partial
+--      ordering of the change bundles.
+--   3. By removing from the smallest bundles, we remove the "least useful"
+--      bundles, maximizing the overall usefulness of our bundles.
+
+-- | Adds a minted value to the change map. This function presumes that the
+-- change map is partially-ordered in ascending order (i.e. the largest
+-- element is at the end of the list).
+--
+-- Properties:
+-- 1. wrt to the total value of assets in the changeMaps, calling this
+-- function is the same as adding the mint value from the change maps:
+--    F.fold changeMaps <> TokenMap.singleton assetId qty
+--          == F.fold (f (assetId, qty) changeMaps)
+-- 2. length is conserved
+-- 3. order is preserved (i.e., the result is in ascending partial order).
+addMintValueToChangeMaps
+    :: (AssetId, TokenQuantity)
+    -> NonEmpty TokenMap
+    -> NonEmpty TokenMap
+addMintValueToChangeMaps (assetId, assetQty) = \case
+    -- This is the largest element, add mint value here
+    (x :| []) ->
+        TokenMap.add (TokenMap.singleton assetId assetQty) x :| []
+    -- There is a larget element, recurse to end of list
+    (x :| (y : rest)) ->
+        let
+            xs = y :| rest
+        in
+          x `NE.cons` addMintValueToChangeMaps (assetId, assetQty) xs
```

Comment:
Suggestion: use `Control.Lens.over` and `Control.Lens.last1` here:

```suggestion
 addMintValueToChangeMaps (assetId, assetQty) ms = ms
      & Lens.over last1
          (\m -> TokenMap.adjustQuantity m assetId (<> assetQty))
```

Reply:
Addressed in 400c7c316. I've gone and used the second solution, only because lens isn't already a dependency of cardano-wallet-core.